tests/testMeasureSources.py

```diff
@@ -300,6 +300,125 @@ def testPixelFlags(self):
+        child1.set("centroid_y", parent.get("centroid_y"))
+        child1.setParent(parent.getId())
+        image.set(xCenter, yCenter, (flux1, 0, 0))
+        foot1 = afwDetection.Footprint(afwGeom.Point2I(x0 + xCenter, y0 + yCenter), 0.1)
+        child1.setFootprint(afwDetection.HeavyFootprintF(foot1, image))
+
+        # Second child is fainter, but we want to be able to measure it!
+        child2 = cat.addNew()
+        child2.set("centroid_x", parent.get("centroid_x") + xOffset)
+        child2.set("centroid_y", parent.get("centroid_y") + yOffset)
+        child2.setParent(parent.getId())
+        image.set(xCenter + xOffset, yCenter + yOffset, (flux2, 0, 0))
+        foot2 = afwDetection.Footprint(afwGeom.Point2I(x0 + xCenter + xOffset, y0 + yCenter + yOffset), 0.1)
+        child2.setFootprint(afwDetection.HeavyFootprintF(foot2, image))
+
+        spans = afwDetection.SpanContainerT()
+        for ss in foot1.getSpans():
+            spans.push_back(ss)
+        for ss in foot2.getSpans():
+            spans.push_back(ss)
+        bbox = afwGeom.Box2I()
+        bbox.include(foot1.getBBox())
+        bbox.include(foot2.getBBox())
+        parent.setFootprint(afwDetection.Footprint(spans, bbox))
+
+        exposure = afwImage.makeExposure(image)
+
+        sfm.run(cat, exposure)
+
+        def checkSource(source, baseName, expectedFlux):
+            """Check that we get the expected results"""
+            self.assertEqual(source.get(baseName + "_flux"), expectedFlux)
+            self.assertGreater(source.get(baseName + "_fluxSigma"), 0)
+            self.assertFalse(source.get(baseName + "_flag"))
+
+        # Deblended
+        checkSource(child1, fieldName, flux1)
+        checkSource(child2, fieldName, flux2)
+
+        # Undeblended
+        checkSource(child1, "undeblended_" + fieldName, flux1 + flux2)
+        checkSource(child2, "undeblended_" + fieldName, flux1 + flux2)
+
+        # Apply aperture correction
+        apCorrMap = afwImage.ApCorrMap()
+        apCorrMap[fieldName + "_flux"] = afwMath.ChebyshevBoundedField(
+            image.getBBox(),
+            apCorrValue*np.ones((1, 1), dtype=float)
+        )
+        apCorrMap[fieldName + "_fluxSigma"] = afwMath.ChebyshevBoundedField(
+            image.getBBox(),
+            apCorrValue*np.zeros((1, 1), dtype=float)
+        )
+
+        apCorr.run(cat, apCorrMap)
+
+        # Deblended
+        checkSource(child1, fieldName, flux1*apCorrValue)
+        checkSource(child2, fieldName, flux2*apCorrValue)
+
+        # Undeblended
+        checkSource(child1, "undeblended_" + fieldName, (flux1 + flux2)*apCorrValue)
+        checkSource(child2, "undeblended_" + fieldName, (flux1 + flux2)*apCorrValue)
+
+        self.assertIn(fieldName + "_apCorr", schema)
+        self.assertIn(fieldName + "_apCorrSigma", schema)
+        self.assertIn("undeblended_" + fieldName + "_apCorr", schema)
+        self.assertIn("undeblended_" + fieldName + "_apCorrSigma", schema)
```

Comment:
Would you mind moving this to a new file and a new TestCase?  `testMeasureSources.py` is a bit of a smorgasbord, and while it's out of scope to fix that here I'd rather not make it worse.